static/src/javascripts/projects/commercial/modules/messenger/resize.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+import fastdom from 'lib/fastdom-promise';
+import { register } from 'commercial/modules/messenger';
+
+type Specs = {
+    width?: string,
+    height?: string,
+};
+
+const normalise = (length: string | number): string => {
+    const lengthRegexp = /^(\d+)(%|px|em|ex|ch|rem|vh|vw|vmin|vmax)?/;
+    const defaultUnit = 'px';
+    const matches = String(length).match(lengthRegexp);
+    if (!matches) {
+        return '';
+    }
+    return matches[1] + (matches[2] === undefined ? defaultUnit : matches[2]);
+};
+
+const resize = (
+    specs: Specs,
+    iframe: HTMLElement,
+    adSlot: HTMLElement
+): ?Promise<any> => {
+    if (
+        !specs ||
+        !('height' in specs || 'width' in specs) ||
+        !iframe ||
+        !adSlot
+    ) {
+        return null;
```

Comment:
Sure thing, I'm not sure what this change would affect either. I've had problems in the past with functions that *may* return a promise, and then don't under certain conditions.However, this less of a concern now that we use Flow.